code/modules/mob/living/carbon/brain/MMI.dm

```diff
@@ -41,6 +41,8 @@
 			user << "<span class='warning'>You aren't sure where this brain came from, but you're pretty sure it's a useless brain!</span>"
 			return
 
+		if(!user.unEquip(src))
+			return

```

Comment:
it should be `if(!user.unEquip(O))` , it's the brain we're trying to unequip not the MMI that we are attacking.